src/main/java/core/basesyntax/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final FruitStorage fruitStorage;
+
+    public BalanceOperationHandler(FruitStorage fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        fruitStorage.getFruits().put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Consider using `fruitStorage.getFruits().put(transaction.getFruit(), fruitStorage.getFruits().getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());` to ensure that the balance operation correctly updates the quantity by adding to any existing amount, rather than overwriting it.